packages/indicators/src/__tests__/builders/buildArithmetic/buildArithmetic.test.ts

```diff
@@ -0,0 +1,174 @@
+    const testData: [string, ArithmeticConfig, AnalyticValue[]][] = [
+      [
+        'simple calculation - integer',
+        {
+          formula: 'A_TO_2019 + B_TO_2019',
+          aggregation: aggregationsByCode(['A_TO_2019', 'B_TO_2019']),
+        },
+        [{ organisationUnit: 'TO', period: '2019', value: 3 }],
+      ],
+      [
+        'simple calculation - float',
+        {
+          formula: 'A_TO_2019 / B_TO_2019',
+          aggregation: aggregationsByCode(['A_TO_2019', 'B_TO_2019']),
+        },
+        [{ organisationUnit: 'TO', period: '2019', value: 0.5 }],
+      ],
+      [
+        'complex calculation',
+        {
+          formula: '((A_TO_2019 + B_TO_2019) * C_TO_2019) / (D_TO_2019 - E_TO_2019)',
+          aggregation: aggregationsByCode([
+            'A_TO_2019',
+            'B_TO_2019',
+            'C_TO_2019',
+            'D_TO_2019',
+            'E_TO_2019',
+          ]),
+        },
+        [{ organisationUnit: 'TO', period: '2019', value: -9 }],
+      ],
+      [
+        'division with zero',
+        {
+          formula: 'A_TO_2019 / (A_TO_2019 + B_TO_2019 - C_TO_2019)',
+          aggregation: aggregationsByCode(['A_TO_2019', 'B_TO_2019', 'C_TO_2019']),
+        },
+        [{ organisationUnit: 'TO', period: '2019', value: Infinity }],
+      ],
+      [
+        'some data elements are undefined in the orgUnit/period combo',
+        {
+          formula: 'A_TO_2020 + UNDEFINED_TO_2019',
+          aggregation: aggregationsByCode(['A_TO_2020', 'UNDEFINED_TO_2019']),
+        },
+        [],
+      ],
+    ];
+
+    it.each(testData)('%s', (_, config, expected) =>
+      expect(
+        buildArithmetic({ aggregator, config, fetchOptions: {} }),
+      ).resolves.toIncludeSameMembers(expected),
+    );
+  });
+
+  describe('calculations for multiple orgUnit/period combos', () => {
+    const testData: [string, ArithmeticConfig, AnalyticValue[]][] = [
+      [
+        'all data elements are defined in all combos',
+        {
+          formula: 'A_TOPG_201920 + B_TOPG_201920',
+          aggregation: aggregationsByCode(['A_TOPG_201920', 'B_TOPG_201920']),
+        },
+        [
+          { organisationUnit: 'TO', period: '2019', value: 3 },
+          { organisationUnit: 'TO', period: '2020', value: 30 },
+          { organisationUnit: 'PG', period: '2019', value: 3.3000000000000003 },
+          { organisationUnit: 'PG', period: '2020', value: 33 },
+        ],
+      ],
+      [
+        'some data elements are undefined in some periods',
+        {
+          formula: 'A_TOPG_201920 + C_TOPG_2019',
+          aggregation: aggregationsByCode(['A_TOPG_201920', 'C_TOPG_2019']),
+        },
+        [
+          { organisationUnit: 'TO', period: '2019', value: 4 },
+          { organisationUnit: 'PG', period: '2019', value: 4.4 },
+        ],
+      ],
+      [
+        'some data elements are undefined in some org units',
+        {
+          formula: 'A_TOPG_201920 + D_TO_201920',
```

Comment:
While this is very concise, I find it hard to know which data elements are undefined without going back to the fixtures. I'm not sure exactly what the solution is to that, and maybe it's a generic problem that we generally expect is unavoidable, but perhaps more descriptive codes could help? I don't know if it's worth it.

Reply:
I thought the codes would help...

`A_TOPG_201920 ` => defined in `TO` and `PG`, for `2019` and `2020`
`D_TO_201920` => defined in `TO`, for `2019` and `2020`

Also `A`, `B`, `C` etc correspond to values including 1, 2, 3 etc

But obviously this wasn't 100% clear for the reader. Perhaps I should add a comment? Or we could count on using similar conventions in the future, making people more familiar with this syntax?

Reply:
Haha well it is clear now you point it out!

Perhaps just add that same note in the `fixtures`, so that the pattern is more obvious.